### PR TITLE
feat: Switch widget

### PR DESCRIPTION
**Pull request by JoaquinBattilana**
## General Changes

- Fixes XYZ bug
- Adds XYZ feature
- …

## Developer Notes

Add any notes here that may be helpful for reviewers.

---

## Author Checklist

Please ensure you, the author, have gone through this checklist to ensure there is an efficient workflow for the reviewers.

- [ ]  The base branch is set to `main`
- [ ]  The title is using [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) formatting
- [ ]  The Github issue has been linked to the PR in the Development section
- [ ]  The General Changes section has been filled out
- [ ]  Developer Notes have been added (optional)

**If the PR is ready for review:**

- [ ]  The PR is in `Open` state and not in `Draft` mode
- [ ]  The `Ready for Dev Review` label has been added

## Reviewer Checklist

Please ensure you, as the reviewer(s), have gone through this checklist to ensure that the code changes are ready to ship safely and to help mitigate any downstream issues that may occur.

- [ ]  End-to-end tests are passing without any errors
- [ ]  Code style generally follows existing patterns
- [ ]  Code changes do not significantly increase the application bundle size
- [ ]  If there are new 3rd-party packages, they do not introduce potential security threats
- [ ]  If there are new environment variables being added, they have been added to the `.env.example` file as well as the pertinant `.github/actions/*` files
- [ ]  There are no CI changes, or they have been approved by the DevOps and Engineering team(s)
- [ ]  Code changes have been quality checked in the ephemeral URL
- [ ]  QA verification has been completed
- [ ]  There are two or more approvals from the core team
- [ ]  Squash and merge has been checked

### CODE REVIEW

**Comment by MareskoY** [comment]
@JoaquinBattilana found few bugs 

**Bug1:** 

select from contains assets with 0. values, don;t need to show 0 lines  in from select

Screenshot 2023-09-21 at 13 10 22

**Bug2:**

it's able to select sameassets pairs like USDC to USDC, this makes no sense 

Screenshot 2023-09-21 at 13 13 07

**Bug3:** 

Switch doesn't work  for USDC => DAI (Polygon) and for some another pairs 

on click switch i get 
```
cannot estimate gas; transaction may fail or may require manual gas limit (error={"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\":\"2.0\",\"id\":98,\"error\":{\"code\":3,\"message\":\"execution reverted: Permit failed\",\"data\":\"0x08c379a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000d5065726d6974206661696c656400000000000000000000000000000000000000\"}}\n","error":{"code":3,"data":"0x08c379a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000d5065726d6974206661696c656400000000000000000000000000000000000000"},"requestBody":"{\"method\":\"eth_estimateGas\",\"params\":[{\"gasPrice\":\"0x199c82cc00\",\"value\":\"0x0\",\"from\":\"0xe4217040c894e8873ee19d675b6d0eec992c2c0d\",\"to\":\"0xdef171fe48cf0115b1d80b88dc8eab59176fee57\",\"data\":\"0x54e3f31b00000000000000000000000000000000000000000000000000000000000000200000000000000000000000002791bca1f2de4661ed88a30c99a7a9449aa841740000000000000000000000008f3cf7ad23cd3cadbd9735aff958023239c6a06300000000000000000000000000000000000000000000000000000000000186a000000000000000000000000000000000000000000000000001622f851029ec9800000000000000000000000000000000000000000000000001628a483514b26700000000000000000000000000000000000000000000000000000000000001e00000000000000000000000000000000000000000000000000000000000000220000000000000000000000000000000000000000000000000000000000000034000000000000000000000000000000000000000000000000000000000000003a000000000000000000000000000000000000000000000000000000000000000000000000000000000000000009abf798f5314bfd793a9e57a654bed35af4a1d60010000000000000000000000000000000000000000000000000000000003138800000000000000000000000000000000000000000000000000000000000003e000000000000000000000000000000000000000000000000000000000650c41732f6c65165ece4bc2949eb5eb63b2195a000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000f3938337f7294fef84e9b2c6d548a93f956cc28100000000000000000000000000000000000000000000000000000000000000e491a32b690000000000000000000000002791bca1f2de4661ed88a30c99a7a9449aa8417400000000000000000000000000000000000000000000000000000000000186a00000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000001000000000000000000004de4f04adbf75cdfc5ed26eea4bbbb991db002036bdd000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e40000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000e4217040c894e8873ee19d675b6d0eec992c2c0d000000000000000000000000216b4b4ba9f3e719726886d34a177484278bfcae00000000000000000000000000000000000000000000000000000000000186a000000000000000000000000000000000000000000000000000000000650c4173000000000000000000000000000000000000000000000000000000000000001c14b5077bada3875ad82a3540881dd55a9412c350ecb9f22c80ecd860a3b36a4f5b09060b5f849c4d40455466f4263d54a68a9e6999e000e2ecb4c83b311ee433\"}],\"id\":98,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"https://rpc-mainnet.matic.quiknode.pro"}, method="estimateGas", transaction={"from":"0xE4217040c894e8873EE19d675b6d0EeC992c2c0D","gasPrice":{"type":"BigNumber","hex":"0x199c82cc00"},"to":"0xDEF171Fe48CF0115B1d80b88dc8eAB59176FEe57","value":{"type":"BigNumber","hex":"0x00"},"data":"0x54e3f31b00000000000000000000000000000000000000000000000000000000000000200000000000000000000000002791bca1f2de4661ed88a30c99a7a9449aa841740000000000000000000000008f3cf7ad23cd3cadbd9735aff958023239c6a06300000000000000000000000000000000000000000000000000000000000186a000000000000000000000000000000000000000000000000001622f851029ec9800000000000000000000000000000000000000000000000001628a483514b26700000000000000000000000000000000000000000000000000000000000001e00000000000000000000000000000000000000000000000000000000000000220000000000000000000000000000000000000000000000000000000000000034000000000000000000000000000000000000000000000000000000000000003a000000000000000000000000000000000000000000000000000000000000000000000000000000000000000009abf798f5314bfd793a9e57a654bed35af4a1d60010000000000000000000000000000000000000000000000000000000003138800000000000000000000000000000000000000000000000000000000000003e000000000000000000000000000000000000000000000000000000000650c41732f6c65165ece4bc2949eb5eb63b2195a000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000f3938337f7294fef84e9b2c6d548a93f956cc28100000000000000000000000000000000000000000000000000000000000000e491a32b690000000000000000000000002791bca1f2de4661ed88a30c99a7a9449aa8417400000000000000000000000000000000000000000000000000000000000186a00000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000001000000000000000000004de4f04adbf75cdfc5ed26eea4bbbb991db002036bdd000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e40000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000e4217040c894e8873ee19d675b6d0eec992c2c0d000000000000000000000000216b4b4ba9f3e719726886d34a177484278bfcae00000000000000000000000000000000000000000000000000000000000186a000000000000000000000000000000000000000000000000000000000650c4173000000000000000000000000000000000000000000000000000000000000001c14b5077bada3875ad82a3540881dd55a9412c350ecb9f22c80ecd860a3b36a4f5b09060b5f849c4d40455466f4263d54a68a9e6999e000e2ecb4c83b311ee433","accessList":null}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.5.3)
```

Screenshot 2023-09-21 at 13 15 58

**Bug4:**
Approve for DAI=>LINK (Polygon)

Errors before approve:  

```
next-dev.js:25 missing revert data in call exception (error={"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\":\"2.0\",\"id\":47,\"error\":{\"code\":-32000,\"message\":\"execution reverted\"}}","error":{"code":-32000},"requestBody":"{\"method\":\"eth_call\",\"params\":[{\"to\":\"0x8f3cf7ad23cd3cadbd9735aff958023239c6a063\",\"data\":\"0xb9844d8d000000000000000000000000e4217040c894e8873ee19d675b6d0eec992c2c0d\"},\"latest\"],\"id\":47,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"https://polygon-rpc.com"}, data="0x", code=CALL_EXCEPTION, version=providers/5.5.3)
```

Error after approve:
```
next-dev.js:25 missing revert data in call exception (error={"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\":\"2.0\",\"id\":47,\"error\":{\"code\":-32000,\"message\":\"execution reverted\"}}","error":{"code":-32000},"requestBody":"{\"method\":\"eth_call\",\"params\":[{\"to\":\"0x8f3cf7ad23cd3cadbd9735aff958023239c6a063\",\"data\":\"0xb9844d8d000000000000000000000000e4217040c894e8873ee19d675b6d0eec992c2c0d\"},\"latest\"],\"id\":47,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"https://polygon-rpc.com"}, data="0x", code=CALL_EXCEPTION, version=providers/5.5.3)
```


**Bug5:** 

for some pairs we get different behavior, approve button not exist but switch button is disabled 

also console errors:
```
Access to fetch at 'https://aave-api-v2.aave.com/addresses/status?address=0xe4217040c894e8873ee19d675b6d0eec992c2c0d' from origin 'http://localhost:3000' has been blocked by CORS policy: The 'Access-Control-Allow-Origin' header has a value 'https://app.aave.com' that is not equal to the supplied origin. Have the server send the header with a valid value, or, if an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

```
useAddressAllowed.tsx:20     GET https://aave-api-v2.aave.com/addresses/status?address=0xe4217040c894e8873ee19d675b6d0eec992c2c0d net::ERR_FAILED 200 (OK)
```
Screenshot 2023-09-21 at 13 25 28


**Bug6:** 

If we change network from defoult(opened market) we get affect whern approve button doesn;t work at all 

Steps: 
1. open app on Eth market
2. open swap widget
3. change polygon
4. choose any pair 
5. try to approve 

Result:
nothing happened also tghere is no concole errors

dif-market

File: src/components/transactions/GasStation/GasStation.tsx [hunk 1]
```diff
@@ -42,28 +46,36 @@ export const GasStation: React.FC<GasStationProps> = ({
...
 }) => {
-  const {
-    state,
-    gasPriceData: { data },
-  } = useGasStation();
-
-  const { walletBalances } = useWalletBalances();
+  const { state } = useGasStation();
+  const currentChainId = useRootStore((store) => store.currentChainId);
+  const selectedChainId = chainId ?? currentChainId;
```
Comment (foodaka) [inline src/components/transactions/GasStation/GasStation.tsx:53]: just a check that ?? only check against `null` and `undefined`.

File: src/components/transactions/Switch/SwitchModalContent.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+          }}
+        />
+      )}
+      <Box sx={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center' }}>
+        <NetworkSelector
+          networks={supportedNetworks}
+          selectedNetwork={selectedChainId}
+          setSelectedNetwork={setSelectedChainId}
+        />
+        <SwitchSlippageSelector slippage={slippage} setSlippage={setSlippage} />
```
[the remark this answers is not in the document]
Comment (JoaquinBattilana) [reply src/components/transactions/Switch/SwitchModalContent.tsx:112]: The design is different from the current one

File: src/ui-config/marketsConfig.tsx [hunk 3]
```diff
@@ -43,6 +43,7 @@ export type MarketDataType = {
     permissions?: boolean;
     debtSwitch?: boolean;
     withdrawAndSwitch?: boolean;
+    switch?: boolean;
```
[the remark this answers is not in the document]
Comment (JoaquinBattilana) [reply src/ui-config/marketsConfig.tsx:46]: I did it per market to allow the fetching of tokens, i think for now that should be okay, but i agree, in the future when we do a list of curated tokens we should do it per network

File: src/components/transactions/Switch/SwitchModalContent.tsx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+            assets={reserves}
+            value={inputAmount}
+            onChange={handleInputChange}
+            usdValue={sellRates?.srcUSD || '0'}
+            symbol={selectedInputReserve?.symbol}
+            onSelect={setSelectedInputReserve}
+            inputTitle={' '}
+          />
+          <AssetInput
+            assets={reserves}
```
[the remark this answers is not in the document]
Comment (JoaquinBattilana) [reply src/components/transactions/Switch/SwitchModalContent.tsx:128]: I added the sort by balance, but the GHO automatically selection should be already working 
```
  const [selectedOutputReserve, setSelectedOutputReserve] = useState(() => {
    const gho = reserves.find((reserve) => reserve.symbol === 'GHO');
    if (gho) return gho;
    return reserves[1];
  });
  ```
  
  I always get GHO when i gho to Ethereum, did you found a case that GHO is not being selected?

File: src/components/transactions/GasStation/GasStation.tsx (hunk 1 above)
Comment (JoaquinBattilana) [reply src/components/transactions/GasStation/GasStation.tsx:53]: It's correct, if the user doesn't pass a chainId(undefined), it'll use the current chain id

File: src/components/transactions/Switch/SwitchModalContent.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (grothem) [reply src/components/transactions/Switch/SwitchModalContent.tsx]: GHO is selected by default, but it's not first in the list

File: src/components/transactions/Switch/SwitchModalContent.tsx [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+  });
+
+  if (sellRates && switchTxState.success) {
+    return (
+      <SwitchTxSuccessView
+        txHash={switchTxState.txHash}
+        amount={inputAmount}
+        symbol={selectedInputReserve.symbol}
+        outSymbol={selectedOutputReserve.symbol}
+        outAmount={normalize(sellRates.destAmount, sellRates.destDecimals)}
```
Comment (grothem) [inline src/components/transactions/Switch/SwitchModalContent.tsx:89]: Would be good to also apply the slippage here, so it'll show the minimum amount received

File: src/components/transactions/Switch/SwitchModalContent.tsx [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+export const SwitchModalContent = ({
+  supportedNetworks,
+  selectedChainId,
+  setSelectedChainId,
+  reserves,
+  selectedNetworkConfig,
+}: SwitchModalContentProps) => {
+  const [slippage, setSlippage] = useState('0.001');
+  const [inputAmount, setInputAmount] = useState('0');
```
Comment (grothem) [inline src/components/transactions/Switch/SwitchModalContent.tsx:46]: With amount inputs we usually default to empty string, then the placeholder value will be shown.

Might have to adjust the paraswap hook to not fetch if the amount is an empty string.

**Comment by MareskoY** [comment]
@JoaquinBattilana  all previous bugs was fixed, but there are a few new small issues that need attention:

Bug 1:
When all numbers are deleted from the input, it sets the value to 0.

Steps to reproduce:
1. Open the switch tool.
2. Enter 222.
3. Remove each number one by one.

Result:
- The input value is set to 0.

[Bug 1 Screenshot]([https://github.com/aave/interface/assets/18032275/afb5f996-d3c2-4e33-b3fb-3db8abc7a751 ](https://github.com/aave/interface/assets/18032275/afb5f996-d3c2-4e33-b3fb-3db8abc7a751))

Bug 2:
Some asset icons, particularly most Avalanche assets, are missing on the success screen.

Bug 2 Screenshot)

Bug 3:
If an error occurs during any step (e.g., approve), the error message persists until a successful transaction or approval is made.

Reproducible case:
1. Choose USDT to any asset.
2. Cancel the MM (money market) approve process.
3. Switch to another market or choose another asset pair.

Result:
- The error message continues to be displayed.

Bug 4:
This is not a technical bug, but it creates a bad user experience. When the modal is opened and the pair is set from ETH to GHO, we are unable to choose ETH in the "switch to" dropdown.

**Review by foodaka — APPROVED** [review]

**Review by grothem — APPROVED** [review]

**Review by foodaka — APPROVED** [review]

**Pull request merged**